source/common/tracing/zipkin/util.cc

```diff
@@ -0,0 +1,84 @@
+#include <iomanip>
+
+#include "rapidjson/writer.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/document.h"
+
+#include "common/common/hex.h"
+#include "common/tracing/zipkin/util.h"
+
+namespace Zipkin {
+
+void Util::mergeJsons(std::string& target, const std::string& source,
+                      const std::string& field_name) {
+  rapidjson::Document target_doc, source_doc;
+  target_doc.Parse(target.c_str());
+  source_doc.Parse(source.c_str());
+
+  target_doc.AddMember(rapidjson::StringRef(field_name.c_str()), source_doc,
+                       target_doc.GetAllocator());
+
+  rapidjson::StringBuffer sb;
+  rapidjson::Writer<rapidjson::StringBuffer> w(sb);
+  target_doc.Accept(w);
+  target = sb.GetString();
+}
+
+void Util::addArrayToJson(std::string& target, const std::vector<const std::string*>& json_array,
+                          const std::string& field_name) {
+  std::string stringified_json_array = "[";
+
+  if (json_array.size() > 0) {
+    stringified_json_array += *(json_array[0]);
+    for (auto it = json_array.begin() + 1; it != json_array.end(); it++) {
+      stringified_json_array += ",";
+      stringified_json_array += **it;
+    }
+  }
+  stringified_json_array += "]";
+
+  mergeJsons(target, stringified_json_array, field_name);
+}
+
+uint64_t Util::timeSinceEpochMicro() {
+  return std::chrono::duration_cast<std::chrono::microseconds>(
+             std::chrono::system_clock::now().time_since_epoch()).count();
+}
+
+uint64_t Util::timeSinceEpochNano() {
+  return std::chrono::duration_cast<std::chrono::nanoseconds>(
+             std::chrono::system_clock::now().time_since_epoch()).count();
+}
+
+uint64_t Util::generateRandom64() {
```

Comment:
I have modified the code to optionally take a custom random-number generated. The code in PR #700 will make sure to call tracer.setRandomGenerator and pass the random-generator currently used by Envoy.